src/Var.cc

```diff
@@ -429,14 +429,25 @@ void add_type(ID* id, TypePtr t, std::unique_ptr<std::vector<AttrPtr>> attr)
 	{
 	std::string new_type_name = id->Name();
 	std::string old_type_name = t->GetName();
+
 	TypePtr tnew;
 
 	if ( (t->Tag() == TYPE_RECORD || t->Tag() == TYPE_ENUM) && old_type_name.empty() )
 		// An extensible type (record/enum) being declared for first time.
 		tnew = std::move(t);
 	else
+		{
+		// If the old type is an error or the old type doesn't exist, then return
+		// an error instead of trying to clone it.
+		if ( t->Tag() == TYPE_ERROR && t->InternalType() == TYPE_INTERNAL_ERROR )
```

Comment:
The comment's "or" doesn't seem to match the condition's `&&`? Maybe it doesn't need to, but then I'd be wondering if we need to check both tag and internal type?

Reply:
We can't just check the tag, since there are cases where tag is set to error but the internal type is set to something valid. Would it be sufficient to just check the internal type for an error?

Reply:
Ah, then I'd think checking just the tag (not the internal type) would give us what we want (because `SetError()` updates the tag, but not the internal type). But sounds like you're saying that's not actually the right thing here?

Either way, fine to leave as you have it if that's working as desired, not worth the trouble.

Reply:
One of the enum btests was failing if I just used the tag.